From cb4289f946f00867dd898b8f87472a24b5fc9313 Mon Sep 17 00:00:00 2001
From: Harunobu Kurokawa <harunobu.kurokawa.dn@renesas.com>
Date: Mon, 31 Oct 2016 12:50:43 +0900
Subject: [PATCH 1359/2066] Revert "PCI: rcar: Remove unused platform data"

This reverts commit d9c64c500aecfc1623a9f21e5a563916d4730832.

 In S2RAM code, pcie driver is extracted from private data of device in
rcar_pcie_suspend(). It needs to add pcie as private data for device.

Reported-by: Hien Dang <hien.dang.eb@renesas.com>
Signed-off-by: Harunobu Kurokawa <harunobu.kurokawa.dn@renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/pci/host/pcie-rcar.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/pci/host/pcie-rcar.c b/drivers/pci/host/pcie-rcar.c
index 7c0e8d6..f4d8d1a 100644
--- a/drivers/pci/host/pcie-rcar.c
+++ b/drivers/pci/host/pcie-rcar.c
@@ -1160,6 +1160,7 @@ static int rcar_pcie_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	pcie->dev = dev;
+	platform_set_drvdata(pdev, pcie);
 
 	INIT_LIST_HEAD(&pcie->resources);
 
-- 
1.7.5.4

